schema/me/conversation.js

```diff
@@ -1,11 +1,131 @@
+import date from "../fields/date"
 import impulse from "../../lib/loaders/impulse"
 import gravity from "../../lib/loaders/gravity"
+import { get } from "lodash"
+import { GraphQLBoolean, GraphQLList, GraphQLObjectType, GraphQLString, GraphQLNonNull, GraphQLEnumType } from "graphql"
+import { ArtworkType } from "../artwork"
+const { IMPULSE_APPLICATION_ID } = process.env
 
-import { GraphQLString, GraphQLNonNull } from "graphql"
+export const BuyerOutcomeTypes = new GraphQLEnumType({
+  name: "BuyerOutcomeTypes",
+  values: {
+    PURCHASED: {
+      value: "purchased",
+    },
+    STILL_CONSIDERING: {
+      value: "still_considering",
+    },
+    HIGH_PRICE: {
+      value: "high_price",
+    },
+    LOST_INTEREST: {
+      value: "lost_interest",
+    },
+    WORK_UNAVAILABLE: {
+      value: "work_unavailable",
+    },
+    OTHER: {
+      value: "other",
+    },
+    BLANK: {
+      value: "",
+    },
+  },
+})
 
-import { ConversationType } from "./conversations"
+export const MessageType = new GraphQLObjectType({
+  name: "MessageType",
+  description: "A message in a conversation.",
+  fields: {
+    id: {
+      description: "Impulse id.",
+      type: GraphQLString,
+    },
+    from_email_address: {
+      description: "Email address of sender.",
+      type: GraphQLString,
+    },
+    snippet: {
+      description: "A snippet of the full message",
+      type: GraphQLString,
+    },
+  },
+})
 
-const { IMPULSE_APPLICATION_ID } = process.env
+export const ConversationFields = {
+  id: {
+    description: "Impulse id.",
+    type: GraphQLString,
+  },
+  inquiry_id: {
+    description: "Gravity inquiry id.",
+    type: GraphQLString,
+  },
+  from_id: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_type: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_name: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  from_email: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_id: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_type: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
+  to_name: {
+    type: new GraphQLNonNull(GraphQLString),
+  },
```

Comment:
Lastly, I’m also wondering what to and from actually means, in the context of a conversation. I thought it was possible for people to join and leave the conversation and that only messages in a conversation have to/from data, just like with normal email threads? /cc @ashkan18